client/src/components/body/profilepage/components/GroupForm.js

```diff
@@ -29,152 +27,126 @@ const GroupForm = (props) => {
+        }
+
+        if (selectedLabs.length >= 0) {
+          for (const lab in checkedLabs) {
+            if (Array.isArray(assignedLabs) && assignedLabs.length !== 0) {
+              if (
+                !assignedLabs.includes(lab.labID) ||
+                assignedLabs.length === 0
+              ) {
+                await GroupService.addGroupLab(groupID, `${lab}`);
+              }
+            } else {
+              await GroupService.addGroupLab(groupID, `${lab}`);
             }
-          });
-          labs.forEach((labID) => {
-            if (!labsAssigned.includes(labID)) {
-              GroupService.addGroupLab(groupID, labID).then(() =>
-                setInstrGroupsUpdated(true),
-              );
+          }
+        }
+
+        if (Array.isArray(assignedLabs) && assignedLabs.length >= 0) {
+          assignedLabs.forEach((lab) => {
+            if (!selectedLabs.includes(lab.labID)) {
+              GroupService.deleteGroupLab(groupID, lab.labID);
             }
           });
-          setInstrGroupsUpdated(true);
         }
-        break;
-      default: // this is the case for enrolling in a group
-        console.log("Group Form Default Case");
+      }
+      setInstrGroupsUpdated(true);
+      props.toggle();
+    } catch (error) {
+      console.error("Error in form submission:", error);
     }
-
-    // Always toggle the modal
-    toggle();
   };
 
-  useEffect(() => {
-    LabService.getAllLabs().then((data) => {
-      setLabs(data);
-    });
-  }, [setState]);
-
-  switch (addMode) {
-    case "add_instr_grp":
-      return (
-        <Form onSubmit={onFormSubmit}>
-          <ModalBody>
-            <FormGroup>
-              <Label for="groupName">Group name</Label>
-              <Input
-                type="text"
-                name="groupName"
-                id="groupName"
-                placeholder={"SWEN 344 Web Engineering Fall 2021"}
-              />
-            </FormGroup>
-            <FormGroup check>
-              <Label for="assign-lab">Choose labs to assign</Label>
-              {labs.map((lab) => (
-                <FormCheckbox isChecked={false} lab={lab} />
-              ))}
-            </FormGroup>
-          </ModalBody>
-          <ModalFooter>
-            <Button color="primary" type="submit">
-              Create Group
-            </Button>
-            <Button color="secondary" onClick={toggle}>
-              Cancel
-            </Button>
-          </ModalFooter>
-        </Form>
-      );
-    case "update_grp_lab":
-      return (
-        <Form onSubmit={onFormSubmit}>
-          <ModalBody>
-            <FormGroup>
-              <Label for="groupName">Group name</Label>
+  return (
+    <Form onSubmit={onFormSubmit}>
+      <ModalBody>
+        <FormGroup>
+          <Label for="groupName">Group Name</Label>
+          <Input
+            type="text"
+            name="groupName"
+            id="groupName"
+            defaultValue={groupName || ""}
```

Comment:
nit: `groupName` will default to `Default Group Name`, so I don't think we need the `|| ""`

Reply:
True. Removed.